### ISSUE TITLE
Feature?: collecting distributed *.md Markdown Documentation 

### ISSUE THREAD

**Issue by SpocWeb** [dotnet/docfx#4750]
**Operation System**: Windows

**DocFX Version Used**: 2.43.0.34314

**Template used**: default

**Steps to Reproduce**:

1. Create a project setup like described in [this Gist](https://gist.github.com/mdschweda/08d82c27d392e6f0a35fbadf0d2e28e7). Host it using docfx docfx_project\docfx.json --serve 

works...

2. Move the index.md File out of the doc/api/ProjectA Folder into the src/ProjectA Folder. Do the same for ProjectB (and C etc.) 

API for ProjectA and ProjectB no longer work 

3. This is unclear: I would like to update the docfx.json so that it retrieves any *.md including the index.md from the respective Project and builds a complete _site from it. 

**Expected Behavior**: I would like to maintain specific *.md Markdown Files within their respective Sub-Projects, but create a _site that not only references the API, but also all Markdown Files from the individual Projects. 

**Actual Behavior**: I succeed to build a Site that contains the collected C# APIs from all Projects in the metadata Section of docfx.json. 
But none of the *.md Files in the ProjectX Folders end up in the docfx_project/api Folders unless I put them there, which makes it impossible to reuse them across different docfx projects. 
Attached is a sample with the Index.md copied just to demo that everything else works. Please notice that the src/ProjectA/ReadMe.md and src/ProjectB/Test.md are NOT available either in doc/docfx_project/_site

I think collecting distributed Markdown Documentation in the respective Project Folders is architecturally a very strong requirement as it facilitates distributed documentation. 

Here is the Project zipped: 

[DocFx-Sample2.zip](https://github.com/dotnet/docfx/files/3297998/DocFx-Sample2.zip)

**Comment by superyyrrzz** [comment]
It is not recommended to put MD files with other .cs/.csproj/.sln. The reason is that we need the source folder structure to determine the output URL. MD is directly built into output HTML, while .cs files are built into YAML files, then output HTML. We can only determine the output from MD+YAML structure. 

It is recommended to use the structure of DocFX's own document: soruce in `src`, MD in `Documentation`. It uses [`"src": "../src"`](https://github.com/dotnet/docfx/blob/1530a2144ed3b693103ac34a6b73f7213e39b301/Documentation/docfx.json#L8) to include source codes.

**Comment by SpocWeb** [comment]
Hi again, just an update how it works for us: 
We did place our *.md conceptual documentation right into our source code, so it is close to the code and will be updated more frequently (avoiding the Effects of [Conways Law](https://en.wikipedia.org/wiki/Conway%27s_law)). 
We had to place the docfx.json, toc.yml and index.md into the root of our repository, but could move all the rest (templates etc.) into a _DocFx subdirectory where we also deploy the _site. 
Maybe you would like to update your documentation. 

The _site works fine, we are still struggling with pdf generation but that is currently postponed.
